bot.py

```diff
@@ -1714,6 +1712,41 @@ async def before_autoupdate(self):
             self.autoupdate.cancel()
             return
 
+    @tasks.loop()
+    async def log_expiry(self):
```

Comment:
Not a showstopper, but this method seems incredibly overcomplicated for what it *needs* to do. I would say it could be trimmed to down to deleting the logs at line 1723 (you get the amount deleted from this so you can log it) and then repeating again after a fixed time, which is how these tasks are typically done anyways. Doing this would make the code much more readable and efficient (though I don't think the extra database calls this currently has would ever actually matter at the scale it's likely to be used at).

Reply:
I don't think it would make it more efficient because instead of running every x minutes and not having any logs to delete, the task will only run when a log needs to be deleted.

Reply:
It's probably going to be mostly irrelevant in efficiency either way (running one query on the database once an hour or day or something that only returns deletion data is basically meaningless, as are basically all the other ones here). I was mostly referring to code readability that you would get by removing all the exit points in this method (and basically removing most of the code) and not duplicating the same query multiple times.

For reference, this is what I'm talking about (untested code):

```python
async def expire_logs(self, log_expire_after):
        now = datetime.utcnow()
        expiration_datetime = now - log_expire_after
        expired_logs = await self.db.logs.delete_many({"closed_at": {"$lte": str(expiration_datetime)}})

        logger.info(f"Deleted {len(expired_logs.deleted_count)} expired logs.")
 ```

imo this is much easier to read and the only functionality you lose is precise scheduling (and you gain being able to easily run it with a command if you wanted)

Reply:
I agree that the majority of the function looks overengineered. I think you can use a constant period loop over the function, and then directly delete from the database based on a query would work better and be more efficient.

We can have the task auto-repeat every hour:
`@tasks.loop(hours=1, reconnect=False)`